### ISSUE TITLE
*DOCKER_API_VERSION* no longer works with *docker-17.09.0-ce*

### ISSUE THREAD

**Issue by spkane** [docker/for-mac#2128]
### Expected behavior

`DOCKER_API_VERSION="1.21" ~/bin/docker-17.09.0-ce -H docker.example.com info`

should return something like this:

```
Containers: 49
 Running: 0
 Paused: 0
 Stopped: 0
Images: 1397
Server Version: 1.9.1
Storage Driver: btrfs
 Build Version: Btrfs v3.19.1
 Library Version: 101
Logging Driver: json-file
Plugins:
 Volume:
 Network:
 Log:
Swarm:
 NodeID:
 Is Manager: false
 Node Address:
Kernel Version: 4.4.30-1.el7.elrepo.x86_64
Operating System: CentOS Linux 7 (Core)
CPUs: 40
Total Memory: 125.7GiB
Name: docker.example.com
ID: LOJ4:3FSK:SZ4F:6X65:J56O:NNWD:LF3Q:ZIMD:S5WV:ATK7:J5IW:WF24
Docker Root Dir: /data/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
Experimental: false
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false

WARNING: No kernel memory limit support
WARNING: No cpu shares support
WARNING: No cpuset support
WARNING: bridge-nf-call-iptables is disabled
WARNING: bridge-nf-call-ip6tables is disabled
```

### Actual behavior

instead it returns this:

```
Error response from daemon: client is newer than server (client API version: 1.24, server API version: 1.21)
```

### Information

Reverting to *docker-17.06.2-ce* causes things to work as expected.

### Steps to reproduce the behavior

  1. Upgrade to *docker-17.09.0-ce*
  2. Run something like this against an older Docker server: `DOCKER_API_VERSION="1.21" ~/bin/docker-17.09.0-ce -H docker.example.com info`

**Comment by thaJeztah** [comment]
Looks related to https://github.com/moby/moby/pull/35008

**Comment by marcuslinke** [comment]
@thaJeztah Will this be fixed or is DOCKER_API_VERSION unsupported from now on?

**Comment by marcuslinke** [comment]
Okay. According to https://github.com/docker/for-mac/issues/2143 seems this is fixed already in 17.10.ce (edge).

**Comment by thaJeztah** [comment]
@marcuslinke correct, it's fixed in versions after 17.09. I opened an internal request to backport the fix to 17.09 (if there will be a 17.09.1 patch release)

**Comment by thaJeztah** [comment]
This should be fixed in 17.09.1 and up
